File: scripts/fix_app_tsx.js
// Fix App.tsx lazy() pattern errors
const fs = require('fs');
const path = 'src/App.tsx';

const content = fs.readFileSync(path, 'utf-8');
console.log(`File size: ${content.length} chars`);
console.log(`Lazy count before: ${(content.match(/lazy\(\(\) =>/g) || []).length}`);
console.log(`Wrong pattern count before: ${(content.match(/as unknown as Promise/g) || []).length}`);

// Pattern to match: lazy(() => import('PATH' as unknown as Promise<{ default: any }>));
const wrongPattern = /lazy\(\(\) => import\('([^']+)' as unknown as Promise<\{ default: any \}\}>\)\);/g;

let count = 0;
const newContent = content.replace(wrongPattern, (match, p1) => {
  count++;
  return `lazy(() => import('${p1}') as any)`;
});

fs.writeFileSync(path, newContent);

console.log(`Replaced ${count} lazy() calls`);
console.log(`Wrong pattern count after: ${(newContent.match(/as unknown as Promise/g) || []).length}`);
console.log(`Correct pattern count after: ${(newContent.match(/as any\)/g) || []).length}`);

File: scripts/fix_app_tsx.py
import re
import sys

with open('src/App.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Match the wrong pattern: lazy(() => import('PATH' as unknown as Promise<{ default: any }>));
wrong = re.compile(r"lazy\(\(\) => import\('([^']+)' as unknown as Promise<\{ default: any \}\}>\)\);")

def replace(m):
    return f"lazy(() => import('{m.group(1)}') as any);"

new_content = wrong.sub(replace, content)

with open('src/App.tsx', 'w', encoding='utf-8') as f:
    f.write(new_content)

print(f"Wrong pattern count after fix: {new_content.count('as unknown as Promise')}")
print(f"Correct pattern count: {new_content.count('as any);')}")
